lib/Driver/Job.cpp

```diff
@@ -106,3 +106,10 @@ void Job::printCommandLine(raw_ostream &os, StringRef Terminator) const {
   printArguments(os, Arguments);
   os << Terminator;
 }
+
+StringRef Job::getCommandLine(StringRef Terminator) const {
+  std::string Out;
+  llvm::raw_string_ostream OS(Out);
+  printCommandLine(OS);
+  return StringRef(Out);
```

Comment:
A StringRef is equivalent to a buf + length. It doesn't extend the lifetime of Out. The StringRef you are returning here will have a pointer to Out's deallocated memory.

I would suggest passing in by reference a llvm::SmallStringImpl and then outside make a llvm::SmallString<128> or something like that. It will be allocated on the stack in most cases.